option.go

```diff
@@ -149,6 +149,18 @@ func NewOption(kind reflect.Kind, names ...string) Option {
 }
 
 func (o *option) WithDefault(v interface{}) Option {
+	if v == nil {
+		panic(fmt.Errorf("cannot use nil as a default"))
+	}
+
+	// if type of value does not match the option type
+	if vKind, oKind := reflect.TypeOf(v).Kind(), o.Type(); vKind != oKind {
+		// if the reason they do not match is not because of Slice vs Array equivalence
+		// Note: Figuring out if the type of Slice/Array matches is not done in this function
+		if !((vKind == reflect.Array || vKind == reflect.Slice) && (oKind == reflect.Array || oKind == reflect.Slice)) {
```

Comment:
This can be removed, but basically is just a hack for StringsOption so that it can still opaquely call this function and everything works. Doesn't seem too unreasonable though.